packages/server/graphql/public/types/Team.ts

```diff
@@ -51,6 +51,16 @@ const Team: TeamResolvers = {
   teamLead: async ({id: teamId}, _args, {dataLoader}) => {
     const teamMembers = await dataLoader.get('teamMembersByTeamId').load(teamId)
     return teamMembers.find((teamMember) => teamMember.isLead)!
+  },
+  isViewerTeamLead: async ({id: teamId}, _args, {authToken, dataLoader}) => {
+    const viewerId = getUserId(authToken)
+    const teamMemberId = toTeamMemberId(teamId, viewerId)
+    const teamMember = await dataLoader.get('teamMembers').load(teamMemberId)
+    return teamMember?.isLead || false
+  },
+  insight: async ({id: teamId}, _args, {dataLoader}) => {
+    const insight = await dataLoader.get('latestInsightByTeamId').load(teamId)
```

Comment:
I'm getting a type error. The problem seems to be that the id is an Int because I'm using SERIAL [here](https://github.com/ParabolInc/parabol/blob/41fc26856c2217bd7b4035d09b6d04b8e9d60ef6/packages/server/postgres/migrations/1722638208553_addInsight.ts). 

Insight.id is defined as type ID [here](https://github.com/ParabolInc/parabol/blob/fc6400af5e00ad56f3b7f8dbd19b38889782ff79/packages/server/graphql/public/typeDefs/Insight.graphql). 

But I get a type error, that's resolved when I do:

```
  insight: async ({id: teamId}, _args, {dataLoader}) => {
    const insight = await dataLoader.get('latestInsightByTeamId').load(teamId)
    if (!insight) return null
    return {
      ...insight,
      id: 'hey'
    }
  }
```

So the problem seems to be the id is an int, but it's expecting a string. I could change the type Insight.id to Int [here](https://github.com/ParabolInc/parabol/blob/fc6400af5e00ad56f3b7f8dbd19b38889782ff79/packages/server/graphql/public/typeDefs/Insight.graphql), but that doesn't seem clean when it's an ID. 

Any ideas?

Reply:
Ah that makes sense, thanks!